src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -100,16 +103,16 @@ public ProvenBlockHeader(PosBlock block)
         public override void ReadWrite(BitcoinStream stream)
         {
             base.ReadWrite(stream);
-            long baseByteSize = stream.ProcessedBytes;
+            this.HeaderSize = stream.ProcessedBytes;
 
             stream.ReadWrite(ref this.merkleProof);
-            this.MerkleProofSize = stream.ProcessedBytes - baseByteSize;
+            this.MerkleProofSize = stream.ProcessedBytes - this.HeaderSize;
```

Comment:
Use `public const int Size = 80;` from base class instead of `this.HeaderSize`
Same for 112, 115